File: Javascript Codewars/6kyu/Exclamation marks series #16: Find the longest successive exclamation marks and question marks combination in the string.js
/*
Description

Find the longest successive exclamation marks and question marks combination in the string. A successive exclamation marks 
and question marks combination must contains two part: a substring of "!" and a substring "?", they are adjacent.

If more than one result are found, return the one which at left side; If no such a combination found, return "".

Examples
find("!!") === ""
find("!??") === "!??"
find("!?!!") === "?!!"
find("!!???!????") === "!!???"
find("!!???!?????") === "!?????"
find("!????!!!?") === "????!!!" 
find("!?!!??!!!?") === "??!!!"

Note
Please don't post issue about difficulty or duplicate.
*/

function find(s) {
  let t = "";

  for (let el of s) {
    if (t === "" || t[t.length - 1] === el) {
      t += el;
    } else {
      t += " " + el;
    }
  }

  t = t.split(" ");
  if (t.length === 1) {
    return "";
  }

  let m = t[0].length + t[1].length;
  let ind = 0;

  for (let i = 1; i < t.length - 1; i++) {
    if (t[i].length + t[i + 1].length > m) {
      m = t[i].length + t[i + 1].length;
      ind = i;
    }
  }

  return t[ind] + t[ind + 1];
}

File: Python Codewars/6kyu/Exclamation marks series #16: Find the longest successive exclamation marks and question marks combination in the string.py
'''
Description

Find the longest successive exclamation marks and question marks combination in the string. 
A successive exclamation marks and question marks combination must contains two part: a 
substring of "!" and a substring "?", they are adjacent.

If more than one result are found, return the one which at left side; If no such a combination found, return "".

Examples
find("!!") === ""
find("!??") === "!??"
find("!?!!") === "?!!"
find("!!???!????") === "!!???"
find("!!???!?????") === "!?????"
find("!????!!!?") === "????!!!" 
find("!?!!??!!!?") === "??!!!"
'''

def find(s):
    t = ''
    
    for el in s:
        if t == '' or t[-1] == el:
            t += el
        else:
            t += " " + el
    
    t = t.split(' ')
    
    if len(t) == 1:
        return ""
    
    m = len(t[0]) + len(t[1])
    ind = 0
    
    for i in range(1, len(t) - 1):
        if len(t[i]) + len(t[i + 1]) > m:
            m = len(t[i]) + len(t[i + 1])
            ind = i
            
    return t[ind] + t[ind + 1]